meteor-app/imports/ui/assessment/assessment-add.js

```diff
@@ -14,16 +14,27 @@ import Congratulations from '/imports/ui/assessment/assessment-congratulations'
 
 import AssessmentAddReview from '/imports/ui/assessment/assessment-add-review'
 
-const mapSchemaToState = schema => (
-  schema
-    .reduce((state, step) => {
-      Object.keys(step.schema.properties)
-        .forEach(prop => {
-          prop === 'services' || prop === 'parts' ? state[prop] = [] : state[prop] = undefined
-        })
-      return state
-    }, {})
-)
+const mapSchemaToState = schema => {
+  // Filter through json form schema to filter out just the form fields
+  return schema.reduce((state, step) => {
+    Object.keys(step.schema.properties)
+      .forEach(prop => {
+        const type = Object.values(step.schema.properties[prop].type).join('')
+        if(type === 'string') {
```

Comment:
Ever used the switch statement? Something like this:
```
switch (type) {
string: state[prop] = ''; break;
integer: state[prop] = 0; break;
boolean: state[prop] = false; break;
}
```
Another way to do it is this:
```
const defaultValues = {
string: '',
integer: 0,
boolean: false,
}
state[prop] = defaultValues[type] || ''
```
I like the last one better, the code is easier to read, and the `defaultValues` object could be moved into the constants file as `CONSTANTS.DEFAUL_VALUES'

Reply:
Ahh.. That is a nice way to do it. Lesson learnt!